packages/react/src/components/Table/TableFoot/TableFoot.jsx

```diff
@@ -50,46 +51,59 @@ const defaultProps = {
 
 const TableFoot = ({
   testId,
-  options: { hasRowExpansion, hasRowSelection, hasRowActions },
+  options: { hasRowExpansion, hasRowSelection, hasRowActions, hasRowNesting },
   tableState: { aggregations, ordering },
   showExpanderColumn,
 }) => {
   const visibleColumns = ordering.filter((col) => !col.isHidden);
 
+  const hasMultiSelect = hasRowSelection === 'multi';
+  const hasExpandOrNest = hasRowExpansion || hasRowNesting;
+  const colSpan = hasExpandOrNest || hasMultiSelect ? 2 : 1;
+
   return (
     <tfoot className={`${iotPrefix}-table-foot`} data-testid={testId}>
-      <TableRow>
-        {hasRowExpansion ? <TableCell key="row-expansion" /> : null}
-        {hasRowSelection === 'multi' ? <TableCell key="row-selection" /> : null}
+      <TableRow key="aggregate-row">
         {visibleColumns.map((orderedCol, index) => {
           const aggregated = aggregations.columns.find((col) => orderedCol.columnId === col.id);
           const isLabelCell = !aggregated && index === 0;
           const cellTestId = `${testId}-aggregation-${orderedCol.columnId}`;
           const cellKey = `${orderedCol.columnId}${index}`;
 
           return isLabelCell ? (
-            <TableCell
-              className={`${iotPrefix}-table-foot--label`}
-              data-testid={cellTestId}
-              key={cellKey}
-            >
-              {aggregations.label}
-            </TableCell>
+            <Fragment key="label-cell-fragment">
+              <TableCell
+                className={`${iotPrefix}-table-foot--label`}
+                data-testid={cellTestId}
+                key={cellKey}
+                colSpan={colSpan}
+              >
+                {aggregations.label}
+              </TableCell>
+              {colSpan === 2 && hasMultiSelect && hasExpandOrNest ? (
```

Comment:
Hmm, if either hasMultiSelect or hasExpandOrNest is true, then I don't think we don't need to test for colSpan === 2. But it is also a bit confusing that we calculate the colspan and at the same time introduce a spacer cell that we just merge.  Is that needed?

Reply:
sure, I can merge it into the colSpan.